retrieval_service/datastore/providers/spanner_gsql_test.py

```diff
@@ -0,0 +1,711 @@
+def db_name() -> str:
+    return get_env_var("DB_NAME", "Spanner Database")
+
+
+@pytest.fixture(scope="module")
+def service_accounts_file_path() -> str:
+    return get_env_var(
+        "SERVICE_ACCOUNT_KEY_FILE", "Service account with permission to spanner"
+    )
+
+
+@pytest.fixture(scope="module")
+async def create_db(
+    db_project: str, db_instance: str, db_name: str, service_accounts_file_path: str
+) -> AsyncGenerator[str, None]:
+    credentials = service_account.Credentials.from_service_account_file(
+        service_accounts_file_path
+    )
+
+    client = spanner.Client(project=db_project, credentials=credentials)
+    instance = client.instance(db_instance)
+
+    database = instance.database(db_name)
+
+    database.create()
+
+    yield db_name
+
+    database.drop()
+    client.close()
+
+
+@pytest_asyncio.fixture(scope="module")
+async def ds(
+    create_db: AsyncGenerator[str, None],
+    db_project: str,
+    db_instance: str,
+    service_accounts_file_path: str,
+) -> AsyncGenerator[datastore.Client, None]:
+    db_name = await create_db.__anext__()
+    cfg = spanner_gsql.Config(
+        kind="spanner-gsql",
+        project=db_project,
+        instance=db_instance,
+        database=db_name,
+        service_account_key_file=service_accounts_file_path,
+    )
+
+    ds = await datastore.create(cfg)
+
+    airports_ds_path = "../../../data/airport_dataset.csv"
+    amenities_ds_path = "../../../data/amenity_dataset.csv"
+    flights_ds_path = "../../../data/flights_dataset.csv"
+    policies_ds_path = "../../../data/cymbalair_policy.csv"
+    airports, amenities, flights, policies = await ds.load_dataset(
+        airports_ds_path,
+        amenities_ds_path,
+        flights_ds_path,
+        policies_ds_path,
+    )
+    await ds.initialize_data(airports, amenities, flights, policies)
+
+    if ds is None:
+        raise TypeError("datastore creation failure")
+
+    yield ds
+
+    await ds.close()
+
+
+async def test_export_dataset(ds: spanner_gsql.Client):
+    airports, amenities, flights, policies = await ds.export_data()
+
+    airports_ds_path = "../../../data/airport_dataset.csv"
+    amenities_ds_path = "../../../data/amenity_dataset.csv"
+    flights_ds_path = "../../../data/flights_dataset.csv"
+    policies_ds_path = "../../../data/cymbalair_policy.csv"
+
+    airports_new_path = "../../../data/airport_dataset.csv.new"
+    amenities_new_path = "../../../data/amenity_dataset.csv.new"
+    flights_new_path = "../../../data/flights_dataset.csv.new"
+    policies_new_path = "../../../data/cymbalair_policy.csv.new"
+
+    await ds.export_dataset(
+        airports,
+        amenities,
+        flights,
+        policies,
+        airports_new_path,
+        amenities_new_path,
+        flights_new_path,
+        policies_new_path,
+    )
+
+    diff_airports = compare(
+        load_csv(open(airports_ds_path), "id"), load_csv(open(airports_new_path), "id")
+    )
+    assert diff_airports["added"] == []
```

Comment:
consolidate these asserts to a function (I just made update to the other test files). Thankyou!